### PR TITLE
 Replace SortedSet with PriorityQueue in Clock 

### PR DESCRIPTION
**Pull request by adamruzicka**
Clock should now scale linearly with the number of events.

```
# bundle exec ruby examples/clock_benchmark.rb       
       user     system      total        real
   100  0.009689   0.003122   0.012811 (  0.022869)
  1000  0.092639   0.011817   0.104456 (  0.096092)
  5000  0.389363   0.040948   0.430311 (  0.399498)
 10000  1.041338   0.084084   1.125422 (  1.023096)
 50000  5.399935   0.637752   6.037687 (  5.478751)
100000 12.343072   1.202518  13.545590 ( 12.339196)
```

### CODE REVIEW

File: lib/dynflow/clock.rb [hunk 1]
```diff
@@ -72,7 +71,8 @@ def tick
     end
 
     def add_timer(timer)
-      @timers.add timer
+      # @timers.add timer
```
Comment (iNecas) [inline lib/dynflow/clock.rb]: please clean out commented code

File: lib/dynflow/clock.rb [hunk 2]
```diff
@@ -84,13 +84,17 @@ def add_timer(timer)
...
+        rescue => e
+          @logger && @logger.error("Failed to apply clock event #{first_timer}, exception: #{e}")
+        end
+        @timers.pop
       end
     end
 
     def first_timer
-      @timers.first
+      @timers.top
```
[the remark this answers is not in the document]
Comment (adamruzicka) [reply lib/dynflow/clock.rb]: The (in our case) minimal element is placed at the end of the underlying array. It should be correct

File: lib/dynflow/clock.rb (hunk 1 above)
Comment (adamruzicka) [reply lib/dynflow/clock.rb]: Why did I even comment this out?

**Comment by adamruzicka** [comment]
I forgot to add the test file to the commit, will push tomorrow

File: examples/clock_benchmark.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  def null
+    @counter += 1
+    @future.success(true) if @counter >= @limit
+  end
+end
+
+def test_case(count)
+  future   = Concurrent.future
```
Comment (iNecas) [inline examples/clock_benchmark.rb]: After update of concurrent-ruby, needs to be changed to `future   = Concurrent::Promises.resolvable_future`

File: examples/clock_benchmark.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+class Receiver
+  def initialize(limit, future)
+    @limit = limit
+    @future = future
+    @counter = 0
+  end
+
+  def null
+    @counter += 1
+    @future.success(true) if @counter >= @limit
```
Comment (iNecas) [inline examples/clock_benchmark.rb]: `s/success/fulfill/`

File: examples/planning_stuck.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env ruby
```
Comment (iNecas) [inline examples/planning_stuck.rb]: Are this additional example files expected here?

File: examples/stuck_singleton.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env ruby
+require_relative 'example_helper'
```
[the remark this answers is not in the document]
Comment (adamruzicka) [reply examples/stuck_singleton.rb]: Gone

**Pull request merged**